### PR TITLE
Bump the minimum PETSc version number.

### PR DESCRIPTION
**Pull request by drwells**
Increasing the minimum version number lets us clean up a bunch of old workarounds.

### CODE REVIEW

File: source/lac/petsc_solver.cc [hunk 1]
```diff
@@ -398,37 +389,10 @@ namespace PETScWrappers
...
-    // like this.
-    /*
-        ierr = KSPGMRESSetRestart (ksp, additional_data.restart_parameter);
-        AssertThrow (ierr == 0, ExcPETScError(ierr));
-    */
-    // so rather expand their macros by hand,
-    // and do some equally nasty stuff that at
-    // least doesn't yield warnings...
-    int (*fun_ptr)(KSP, int);
-    ierr = PetscObjectQueryFunction(reinterpret_cast<PetscObject>(ksp),
```
Comment (tjhei) [inline source/lac/petsc_solver.cc:422]: Yay!

**Review by tjhei — APPROVED** [review]

File: source/lac/petsc_solver.cc (hunk 1 above)
Comment (drwells) [reply source/lac/petsc_solver.cc:422]: I don't know what that used to be but its definitely a normal function in `petscksp.h` these days.

**Review by masterleinad — APPROVED** [review]

**Review by marcfehling — APPROVED** [review]
PETSc 3.7.0 has been released in April 2016. Good idea to move forward!

**Review by tamiko — APPROVED** [review]

**Pull request merged**